syntax = "proto3";

package smartclass;

service SmartClass {
  rpc JoinRoom (JoinRoomRequest) returns (JoinRoomResponse);
  rpc GetQuiz (Player) returns (Quiz);
  rpc SubmitAnswer (Answer) returns (GameStatus);
  rpc ExitRoom (Player) returns (GameStatus);
}

message JoinRoomRequest {
  string player_name = 1;
}

message JoinRoomResponse {
  bool joined = 1;
  string topic = 2;
  int32 game_timer = 3;
  string message = 4;
  int32 quizzes_count = 5;
}

message Player {
  string player_name = 1;
}

message Quiz {
  string question = 1;
  repeated string options = 2;
  string answer = 3;
}

message Answer {
  string player_name = 1;
  bool isCorrect = 2;
}

message GameStatus {
  bool isOver = 1;
  int32 score = 2;
}